### ISSUE TITLE
ECMAFeature classes should throw when disabled

### ISSUE THREAD

**Issue by emmenko** [babel/babel-eslint#53]
I've noticed that if I use `babel-eslint` as parser, the `ecmaFeatures.classes = false` option does nothing.

Given this configuration
```bash
$ cat .eslintrc
/* http://eslint.org/docs/rules/ */
{
  "parser": "babel-eslint",
  "ecmaFeatures": {
    "classes": false
  },
  "env": {
    "node": true
  },
  "rules": {
    "eol-last": 2,
    "quotes": [2, "single"],
    "semi": 0,
    "strict": 0
  }
}
```

and this file
```javascript
// test-class.js

class Foo {
  constructor() {
    this.foo = 'bar'
  }
}

```

When I run

```bash
$ eslint test-class.js
$ echo $?
0
```

When I remove the `parser` option it works
 
```bash
$ cat .eslintrc
/* http://eslint.org/docs/rules/ */
{
  "ecmaFeatures": {
    "classes": false
  },
  "env": {
    "node": true
  },
  "rules": {
    "eol-last": 2,
    "quotes": [2, "single"],
    "semi": 0,
    "strict": 0
  }
}

$ eslint test-class.js

test-class.js
  1:1  error  Unexpected reserved word

✖ 1 problem (1 error, 0 warnings)
```

Is this a problem with `babel-eslint` ?

**Comment by sebmck** [comment]
Nope, intended behaviour. Acorn does not currently support feature-specific flags.

**Comment by emmenko** [comment]
Woah, that was fast ;)

So what should I do? Do you have any suggestion or workaround?

**Comment by sebmck** [comment]
Can't really do anything about it.

**Comment by emmenko** [comment]
So that means that any `ecmaFeatures` option is ignored?

**Comment by sebmck** [comment]
Correct.

**Comment by emmenko** [comment]
Hmm ok so if want to forbid the use of `class` (or something else) I should create an `eslint-plugin` ?

**Comment by sebmck** [comment]
That's the only way, yes.

**Comment by emmenko** [comment]
Ok, at least I know how to proceed. Many thanks for the fastest support ever! ;)

**Comment by emmenko** [comment]
For who's interested, I'm creating a plugin to forbid `class`
https://github.com/emmenko/eslint-plugin-no-class
